app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/ComposeActivity.kt

```diff
@@ -0,0 +1,174 @@
+import androidx.compose.animation.AnimatedVisibility
+import androidx.compose.animation.core.Spring
+import androidx.compose.animation.core.spring
+import androidx.compose.animation.slideInVertically
+import androidx.compose.animation.slideOutVertically
+import androidx.compose.material.Scaffold
+import androidx.compose.material.ScaffoldState
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.material.icons.filled.Menu
+import androidx.compose.material.rememberScaffoldState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.lifecycle.asFlow
+import androidx.navigation.NavController
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.currentBackStackEntryAsState
+import androidx.navigation.compose.rememberNavController
+import com.bottlerocketstudios.brarchitecture.R
+import com.bottlerocketstudios.brarchitecture.domain.utils.MutableStateFlowDelegate
+import com.bottlerocketstudios.compose.navdrawer.NavDrawer
+import com.bottlerocketstudios.compose.navdrawer.NavItemState
+import com.bottlerocketstudios.compose.resources.ArchitectureDemoTheme
+import com.bottlerocketstudios.compose.widgets.AppBar
+import kotlinx.coroutines.launch
+import org.koin.androidx.viewmodel.ext.android.viewModel
+
+class ComposeActivity : ComponentActivity() {
+    val activityViewModel: MainActivityViewModel by viewModel()
+
+    /**
+     *   EMPTY_TOOLBAR_TITLE is used to show toolbar without a title.
+     */
+    companion object {
+        const val EMPTY_TOOLBAR_TITLE = " "
+    }
+
+    // Lazy initialized public interface that provides access to view model
+    val controls by lazy { Controls(activityViewModel) }
+    class Controls(val viewModel: MainActivityViewModel) {
+        var title by MutableStateFlowDelegate(viewModel.title)
+        var topLevel by MutableStateFlowDelegate((viewModel.topLevel))
+    }
+
+    var navIntercept: (() -> Boolean)? = null
+
+    @Composable
+    fun <T : BaseViewModel> T.ConnectBaseViewModel(block: @Composable (T) -> Unit) {
+        // Reset Controls
+        controls.title = ""
+        controls.topLevel = false
+        navIntercept = null
+
+        // Connect external routing to activity
+        launchIO {
+            externalNavigationEvent.asFlow().collect { runOnMain { it.navigate(this@ComposeActivity) } }
+        }
+
+        block.invoke(this)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+
+        super.onCreate(savedInstanceState)
+        setContent {
+            val coroutineScope = rememberCoroutineScope()
+            val navController = rememberNavController()
+            val scaffoldState = rememberScaffoldState()
+            val scope = rememberCoroutineScope()
```

Comment:
Ended up with 2 coroutine scopes.. oops

Reply:
Nice catch! Let's simplify to a single one (seems reasonable - if both are needed then refactor to have better names)